registry/core/service: remove incorrect comments

tklauser opened the pull request:
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

These interface methods are not only used for testing. Remove the incorrect comments.

As suggested by @aojea on https://github.com/cilium/cilium/pull/25498/files#r1196364151

#### Which issue(s) this PR fixes:

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

aojea commented:
/ok-to-test
/lgtm
/approve

used in 
https://github.com/kubernetes/kubernetes/blob/51afa26513543251be2fb3d61fed43b191a0e364/pkg/registry/core/service/ipallocator/controller/repair.go#L243

and in 

https://github.com/kubernetes/kubernetes/blob/b022475448b3589bf881e9d6d255501b9752d143/pkg/registry/core/service/ipallocator/ipallocator.go#L165

aojea commented:
/sig network

aojea commented:
@tklauser the commit message seems to be invalid,I think is because it has references,  just remove the references and indicate that those methods are not longer used for testing or something like that

tklauser committed:
registry/core/service: remove incorrect comments

These interface methods are not only used for testing. Remove the
incorrect comments.
